components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java

```diff
@@ -2000,4 +2017,120 @@ public static SAMLSSOServiceProviderDO getSPConfig(String tenantDomain, String i
+            logoutReqDTO.setAssertionConsumerURL(serviceProviderDO.getAssertionConsumerUrl());
+        }
+
+        LogoutRequest logoutReq = new LogoutRequestBuilder().buildObject();
+
+        logoutReq.setID(SAMLSSOUtil.createID());
+
+        String destination = logoutReqDTO.getAssertionConsumerURL();
+        if (destination != null) {
+            logoutReq.setDestination(destination);
+        }
+
+        DateTime issueInstant = new DateTime();
+        logoutReq.setIssueInstant(issueInstant);
+        logoutReq.setIssuer(SAMLSSOUtil.getIssuerFromTenantDomain(serviceProviderDO.getTenantDomain()));
+        logoutReq.setNotOnOrAfter(new DateTime(issueInstant.getMillis() + 5 * 60 * 1000));
+
+        NameID nameId = new NameIDBuilder().buildObject();
+        nameId.setFormat(serviceProviderDO.getNameIDFormat());
+        nameId.setValue(subject);
+        logoutReq.setNameID(nameId);
+
+        SessionIndex sessionIndex = new SessionIndexBuilder().buildObject();
+        sessionIndex.setSessionIndex(sessionId);
+        logoutReq.getSessionIndexes().add(sessionIndex);
+
+        logoutReq.setReason(SAMLSSOConstants.SingleLogoutCodes.LOGOUT_USER);
+
+        return logoutReq;
+    }
+
+    // this method is to be removed
+    public static byte[] setSignature(String queryString) throws IdentityException, KeyStoreException {
+
+        byte[] signedString = null;
+
+        String keyAlias = ServerConfiguration.getInstance().getFirstProperty(SECURITY_KEY_STORE_KEY_ALIAS);
+        if (StringUtils.isBlank(keyAlias)) {
+            throw new IdentityException("Invalid file configurations. The key alias is not found.");
+        }
+
+        KeyStoreAdmin keyAdmin = null;
+        try {
+            keyAdmin = new KeyStoreAdmin(MultitenantConstants.SUPER_TENANT_ID,
+                    SAMLSSOUtil.getRegistryService().getGovernanceSystemRegistry());
+        } catch (RegistryException e) {
+            log.error("Error in setting keystore admin", e);
+        }
+
+        try {
+            String path = CarbonUtils.getCarbonHome() + "/repository/resources/security/wso2carbon.jks";
+            FileInputStream cert = new FileInputStream(path);
+            KeyStore keyStore = KeyStore.getInstance("JKS");
+            keyStore.load(cert, "wso2carbon".toCharArray());
+            PrivateKey privateKey = (PrivateKey) keyAdmin.getPrivateKey(keyAlias, true);
+
+            java.security.Signature instance = java.security.Signature.getInstance("SHA1withRSA");
+            instance.initSign(privateKey);
+            instance.update(queryString.getBytes());
+            signedString = instance.sign();
+        } catch (IOException | NoSuchAlgorithmException | CertificateException | InvalidKeyException | SignatureException e) {
+            log.error("Error in setting signature.", e);
+        }
+
+        return signedString;
+    }
+
+    public static List<SAMLSSOServiceProviderDO> getOtherSessionParticipants(String sessionId, String issuer) {
```

Comment:
If it is a public, protected or default method, we need a method header comments. (If you can add them for privates, that is even better.)